crypto/asn1/tasn_new.c

```diff
@@ -186,7 +186,7 @@ static int asn1_item_ex_combine_new(ASN1_VALUE **pval, const ASN1_ITEM *it,
         for (i = 0, tt = it->templates; i < it->tcount; tt++, i++) {
             pseqval = asn1_get_field_ptr(pval, tt);
             if (!ASN1_template_new(pseqval, tt))
-                goto memerr;
+                goto auxerr;
```

Comment:
I understand why you made that choice.  Unfortunately, that will give off the wrong error code, so what I'd do if I was you is to change this back and add a `ASN1_item_ex_free(pval, it);` in the `memerr:` code.